satellite/orders/service.go

```diff
@@ -633,39 +634,75 @@ func (service *Service) CreatePutRepairOrderLimits(ctx context.Context, repairer
 		return nil, err
 	}
 
-	limits := make([]*pb.AddressedOrderLimit, totalPieces)
-	var pieceNum int32
-	for _, node := range newNodes {
-		for int(pieceNum) < totalPieces && getOrderLimits[pieceNum] != nil {
-			pieceNum++
+	var limits []*pb.AddressedOrderLimit
+	{ // Create the order limits for being used to upload the repaired pieces
+		redundancy, err := eestream.NewRedundancyStrategyFromProto(pointer.GetRemote().GetRedundancy())
+		if err != nil {
+			return nil, Error.Wrap(err)
 		}
 
-		if int(pieceNum) >= totalPieces { // should not happen
-			return nil, Error.New("piece num greater than total pieces: %d >= %d", pieceNum, totalPieces)
+		totalPieces := redundancy.TotalCount()
+		limits = make([]*pb.AddressedOrderLimit, totalPieces)
+
+		totalPiecesAfterRepair := int(
+			math.Ceil(
+				float64(redundancy.OptimalThreshold()) * (1 + float64(service.repairMaxExcessOptimalThreshold/100)),
```

Comment:
same thing - `float64(service.repairMaxExcessOptimalThreshold/100)` will always be 0 when `repairMaxExcessOptimalThreshold` is less than 100 (see go playground link in other comment)

Reply:
Because the configuration value has been changed from int to float64, this bug doesn't exist anymore